### PR TITLE
[SofaFramework] Various cleaning (non-breaking)

### PR DESCRIPTION
**Pull request by mthieffry**
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaConstraint/FreeMotionAnimationLoop.cpp [hunk 1]
```diff
@@ -303,7 +288,9 @@ void FreeMotionAnimationLoop::step(const sofa::core::ExecParams* params, SReal d
 
 SOFA_DECL_CLASS(FreeMotionAnimationLoop)
 
-int FreeMotionAnimationLoopClass = core::RegisterObject("Constraint solver")
+int FreeMotionAnimationLoopClass = core::RegisterObject(R"(
+The animation loop to use with constraints.
+You must add this loop at the beginning of the scene if you are using constraints.")")
```
Comment (guparan) [inline modules/SofaConstraint/FreeMotionAnimationLoop.cpp:289]: What is `R` in `core::RegisterObject(R"(` ?

File: modules/SofaConstraint/FreeMotionAnimationLoop.cpp [hunk 2]
```diff
@@ -184,20 +173,18 @@ void FreeMotionAnimationLoop::step(const sofa::core::ExecParams* params, SReal d
...
+    simulation::MechanicalVOpVisitor lambdaMultInvDt(params, cparams.lambda(), sofa::core::ConstMultiVecId::null(), cparams.lambda(), 1.0 / dt);
+    lambdaMultInvDt.setMapped(true);
+    this->getContext()->executeVisitor(&lambdaMultInvDt);
+    simulation::MechanicalComputeGeometricStiffness geometricStiffnessVisitor(&mop.mparams, cparams.lambda());
+    this->getContext()->executeVisitor(&geometricStiffnessVisitor);
 
     // Free Motion
-    AdvancedTimer::stepBegin("FreeMotion");
+    {
+    ScopedAdvancedTimer timer("FreeMotion");
```
Comment (guparan) [inline modules/SofaConstraint/FreeMotionAnimationLoop.cpp]: Forgot to indent the new scope ;-)

File: modules/SofaConstraint/FreeMotionAnimationLoop.cpp [hunk 3]
```diff
@@ -127,54 +123,47 @@ void FreeMotionAnimationLoop::step(const sofa::core::ExecParams* params, SReal d
...
-        MultiVecDeriv df(&vop, core::VecDerivId::dforce()); df.realloc(&vop, !d_threadSafeVisitor.getValue(), true);
-    }
+    MultiVecDeriv dx(&vop, core::VecDerivId::dx()); dx.realloc(&vop, !d_threadSafeVisitor.getValue(), true);
+    MultiVecDeriv df(&vop, core::VecDerivId::dforce()); df.realloc(&vop, !d_threadSafeVisitor.getValue(), true);
 
     // This solver will work in freePosition and freeVelocity vectors.
     // We need to initialize them if it's not already done.
-    sofa::helper::AdvancedTimer::stepBegin("MechanicalVInitVisitor");
+    {
+    ScopedAdvancedTimer timer("MechanicalVInitVisitor");
```
Comment (guparan) [inline modules/SofaConstraint/FreeMotionAnimationLoop.cpp]: Forgot to indent the new scope ;-)

**Review by guparan — COMMENTED** [review]
Thanks for the PR! Didn't know we have a RAII AdvancedTimer :+1: 
Here is a kind style review.

File: modules/SofaConstraint/FreeMotionAnimationLoop.cpp (hunk 1 above)
Comment (damienmarchal) [reply modules/SofaConstraint/FreeMotionAnimationLoop.cpp:289]: https://en.cppreference.com/w/cpp/language/string_literal

**Comment by damienmarchal** [comment]
Interesting CI results a test failer I never heard about :)

@guparan , @VannesteFelix  any idea if this is related to the PR ?

**Comment by guparan** [comment]
It is.
The error `File size mismatch` appears because in `WriteState_test::createScene` exported files "particleGravityX.data" and "particleGravityY.data" are different from reference files ""particleGravityX-reference.data" and "particleGravityY-reference.data".
This is due to the change in MechanicalObject adding `std::scientific << std::setprecision(18)`.
Exported files should be like
```
T= 0
  X= 0 0 0
...
```
and are now like
```
T= 0
  X= 0.000000000000000000e+00 0.000000000000000000e+00 0.000000000000000000e+00
...
```

Solution 1: regenerate the references located in `modules/SofaExporter/SofaExporter_test/files`.
Solution 2: revert MechanicalObject change.

**Comment by damienmarchal** [comment]
Hey @mthieffr you still have a bit work before entering the top50 :)

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Pull request merged**